Sources/JsonModel/Documentable.swift

```diff
@@ -424,6 +427,26 @@ public class JsonDocumentBuilder {
         }
     }
     
+    private func addTypeKey(for serializer: DocumentableInterface, _ pointer: KlassPointer) {
+        let typeProperty = type(of: serializer).typeDocumentProperty()
+        guard case .reference(let docType) = typeProperty.propertyType
+        else {
+            return
+        }
+        createPointer(documentableType: docType, parent: pointer, baseUrl: pointer.baseUrl)
+    }
+    
+    @discardableResult
+    private func createPointer(documentableType: Documentable.Type, parent: KlassPointer, baseUrl: URL) -> KlassPointer {
+        // Create a new pointer.
+        let pointer = KlassPointer(klass: documentableType, baseUrl: baseUrl, parent: parent)
+        pointer.modelName = factory?.modelName(for: pointer.className) ?? pointer.className
+        
+        // First add the object in case there is recursive mapping.
```

Comment:
Confused why this says "First" when it is the last thing done in the function...?

Reply:
Ha, ha! b/c I don't read my own comments when refactoring.